frontend/mainwindow.cpp

```diff
@@ -3,7 +3,10 @@
 
 #include "mainwindow.h"
 #include "./ui_mainwindow.h"
+#include "datagathering.cpp"
+#include "filtering.cpp"
 #include "jsontablemodel.h"
+#include "searching.cpp"
```

Comment:
Should never include definition files like this.
```suggestion
#include "jsontablemodel.h"
```